### PR TITLE
MAUI-Prepare the UG documentation for in-build Pull to refresh.

### PR DESCRIPTION
**Pull request by sudarsanMS**
### Description

- Prepare the UG documentation for in-build Pull to refresh.

### CODE REVIEW

File: MAUI/DataGrid/pull-to-refresh.md [hunk 1]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
 
 # Pull To Refresh in MAUI DataGrid (SfDataGrid)
 
+The data grid enables the `PullToRefresh` option by setting the [SfDataGrid.AllowPullToRefresh]() property to `true`, the control supports for refreshing the data source at runtime while doing the pull to refresh action.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:13]: To enable the `PullToRefresh` option in the data grid, set the [SfDataGrid.AllowPullToRefresh]() property to `true`. This allows the control to refresh the data source at runtime when the pull-to-refresh gesture is performed.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 2]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
 
+The data grid enables the `PullToRefresh` option by setting the [SfDataGrid.AllowPullToRefresh]() property to `true`, the control supports for refreshing the data source at runtime while doing the pull to refresh action. 
+
+## Pull to refresh command
+
+The data grid refreshes the data in view at runtime by triggering an `ICommand` bound to the [SfDataGrid.PullToRefreshCommand]() property. When you perform a pull-to-refresh action, if the progress bar reaches 100%, this command is triggered to refresh the records in view.
+
+Set the [SfDataGrid.IsBusy]() property to `true` before refreshing the records to notify the grid that a pull-to-refresh action is being performed, and set the property to `false` after the view is refreshed. You can also adjust the pull-to-refresh animation time by setting a delay in the sample.
+
+To enable and perform the pull-to-refresh operation, follow the code example:
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:21]: The data grid can refresh its data during runtime by executing an `ICommand` bound to the [SfDataGrid.PullToRefreshCommand]() property. When a pull-to-refresh gesture is performed and the progress bar reaches 100%, this command is triggered to update the records in view.

To indicate that a pull-to-refresh operation is in progress, set the [SfDataGrid.IsBusy]() property to true before starting the refresh. Once the refresh is complete, set this property back to false. You can also customize the duration of the pull-to-refresh animation by adding a delay in the code.

To enable and perform the pull-to-refresh operation, refer to the following code example:

File: MAUI/DataGrid/pull-to-refresh.md [hunk 3]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+        });
+    }        
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Pull To Refresh" src="Images\pull-to-refresh\net-maui-pull-to-refresh-slide-on-top.gif" width="404"/>
+
+### Transition Mode
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:68]: Transition mode

File: MAUI/DataGrid/pull-to-refresh.md [hunk 4]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+
+The [TransitionMode]() property specifies the mode of the animations. It has two modes:
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:70]: To customize the progress view animation, adjust the `SfDataGrid.TransitionMode` property. By default, the transition mode is set to `SlideOnTop`.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 5]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+
+<syncfusion:SfDataGrid x:Name="dataGrid"
+                       TransitionType="SlideOnTop"
+                       AllowPullToRefresh="True"
+                       ItemsSource="{Binding OrdersInfo}">
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="TransitionMode SlideOnTop" src="Images\pull-to-refresh\net-maui-pull-to-refresh-slide-on-top.gif" width="404"/>
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:88]: Remove the slide on top code snippets and contents.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 6]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+
+The following code example demonstrates how to set the `TransitionMode` to `Push` in SfPullToRefresh. This transition moves only the refresh content, while the header remains fixed.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:90]: The following code snippet demonstrates how to change the transition mode to `Push`. In this mode, only the scrollable view moves, while the header remains fixed:

File: MAUI/DataGrid/pull-to-refresh.md [hunk 7]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+                       ItemsSource="{Binding OrdersInfo}">
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="TransitionMode Push" src="Images\pull-to-refresh\net-maui-pull-to-refresh-push.gif" width="404"/>
+
+### Customization
+
+The data grid allows you to customize various properties such as TransitionMode, PullingThreshold, ProgressBackground, ProgressColor, and more. Refer to the code example below to customize the ProgressColor and ProgressBackground of the pull-to-refresh view.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:107]: You can customize various properties of `SfPullToRefresh` in the data grid, including `TransitionMode`, `PullingThreshold`, `ProgressBackground`, `ProgressColor`, and more.

File: MAUI/DataGrid/pull-to-refresh.md [hunk 8]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+{% tabs %}
+{% highlight c# %}
+
+public MainPage()
+{
+    InitializeComponent();
+    dataGrid.AllowPullToRefresh = true;
+    dataGrid.TransitionType = PullToRefreshTransitionType.Push;
+    dataGrid.DataGridLoaded += DataGrid_DataGridLoaded;
+}
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:118]: public MainPage()
{
    InitializeComponent();
    dataGrid.AllowPullToRefresh = true;
    dataGrid.DataGridLoaded += DataGrid_DataGridLoaded;
}

File: MAUI/DataGrid/pull-to-refresh.md [hunk 9]
```diff
@@ -10,6 +10,135 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+    if (dataGrid.Children.Count > 0)
+    {
+        SfPullToRefresh refreshView = dataGrid.Children.OfType<SfPullToRefresh>().First();
+        refreshView.ProgressColor = Color.FromArgb("ffb703");
+        refreshView.ProgressBackground = Color.FromArgb("023047");
+        refreshView.ProgressThickness = 4;
+        refreshView.RefreshViewThreshold = 10;
+        refreshView.PullingThreshold = 120;
+    }
+}
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:133]: private void DataGrid_DataGridLoaded(object? sender, EventArgs e)
{
        SfPullToRefresh refreshView = dataGrid.Children.OfType<SfPullToRefresh>().First();
        if (refreshView != null)
        {
            refreshView.ProgressColor = Color.FromArgb("ffb703");
            refreshView.ProgressBackground = Color.FromArgb("023047");
            refreshView.ProgressThickness = 4;
            refreshView.RefreshViewThreshold = 10;
            refreshView.PullingThreshold = 120;
        }
}

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/pull-to-refresh.md (hunk 1 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:13]: modified

File: MAUI/DataGrid/pull-to-refresh.md (hunk 2 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:21]: modified

File: MAUI/DataGrid/pull-to-refresh.md (hunk 3 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:68]: modified

File: MAUI/DataGrid/pull-to-refresh.md (hunk 5 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:88]: removed

File: MAUI/DataGrid/pull-to-refresh.md (hunk 6 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:90]: changed

File: MAUI/DataGrid/pull-to-refresh.md (hunk 8 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:118]: changed

File: MAUI/DataGrid/pull-to-refresh.md [hunk 10]
```diff
@@ -10,6 +10,118 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+{% endtabs %}
+
+<img alt="Pull To Refresh" src="Images\pull-to-refresh\net-maui-pull-to-refresh-slide-on-top.gif" width="404"/>
+
+### Transition mode
+
+To customize the progress view animation, adjust the [SfDataGrid.TransitionMode]() property. By default, the transition mode is set to SlideOnTop.
+
+* [SlideOnTop]()
+* [Push]()
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:74]: Remove the transition modes bullet marks

File: MAUI/DataGrid/pull-to-refresh.md [hunk 11]
```diff
@@ -10,6 +10,118 @@ keywords : maui datagrid, maui grid, grid maui, maui gridview, grid in maui, .ne
...
+                       ItemsSource="{Binding OrdersInfo}">
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="TransitionMode Push" src="Images\pull-to-refresh\net-maui-pull-to-refresh-push.gif" width="404"/>
+
+### Customization
+
+You can customize various properties of SfPullToRefresh in the data grid, including TransitionMode, PullingThreshold, ProgressBackground, ProgressColor, and more.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/pull-to-refresh.md:93]: You can provide API links to the already available properties.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]
@sudarsanMS - Include border to the images and address the review corrections.

File: MAUI/DataGrid/pull-to-refresh.md (hunk 7 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md]: modified

File: MAUI/DataGrid/pull-to-refresh.md (hunk 10 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:74]: removed

File: MAUI/DataGrid/pull-to-refresh.md (hunk 11 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/pull-to-refresh.md:93]: added

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine

**Pull request merged**